kernel/src/fs/pipe.rs

```diff
@@ -204,7 +209,20 @@ impl PipeWriter {
             producer.write_fallible(reader)
         };
 
-        self.state.write_with(write)
+        let res = self.state.write_with(write);
+        if res.is_err_and(|e| e.error() == Errno::EPIPE) {
+            let thread = current_thread!();
+            if let Some(posix_thread) = thread.as_posix_thread() {
+                posix_thread.enqueue_signal(Box::new(UserSignal::new(
+                    SIGPIPE,
+                    UserSignalKind::Kill,
```

Comment:
Is this correct? I don't think this signal is generated by the `kill` system call.

Reply:
> #define SI_USER		0		/* sent by kill, sigsend, raise */

I think the Linux behavior is confusing. I'm not sure why `SIGPIPE` should belong to `SI_USER` either. But anyway, thanks for the confirmation. I think it's okay if it's the exact Linux behavior.